react/components/ModalDialog/index.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { FC, useEffect } from 'react'
+import PropTypes from 'prop-types'
+
+import Modal, { Props as ModalProps } from '../Modal'
+import Button from '../Button'
+
+type DialogOption = {
+  label: string
+  onClick: () => unknown
+}
+
+type Props = Pick<
+  ModalProps,
+  'children' | 'isOpen' | 'onClose' | 'onCloseTransitionFinish' | 'container'
+> & {
+  loading?: boolean
+  confirmation: DialogOption & { isDangerous?: boolean | null }
+  cancelation: DialogOption
+}
+
+const ModalDialog: FC<Props> = ({
+  children,
+  confirmation,
+  cancelation,
+  loading,
+  ...props
+}) => {
+  useEffect(() => {
+    console.warn(
+      'ModalDialog will be deprecated soon, please prefer to use Modal component. https://styleguide.vtex.com/#/Components/Overlays/Modal'
+    )
+  }, [])
+
+  const handleCancelation = () => {
+    cancelation.onClick?.()
+  }
+
+  const handleConfirmation = () => {
+    confirmation.onClick?.()
+  }
```

Comment:
I think you don't need to create these function, you could just pass it to the elements